src/pages/workspaces/workspace/workflows/WorkflowView.test.js

```diff
@@ -1,126 +1,349 @@
+                methodName: 'echo_to_file',
+                sourceRepo: 'agora',
+                methodUri: 'agora://gatk/echo_to_file/12',
+                methodVersion: 12,
+              },
+              rootEntityType: 'sra',
+              name: 'echo_to_file-configured',
             }),
+            save: jest.fn().mockReturnValue(mockSave),
+            launch: jest.fn().mockReturnValue(launchPayload),
           }),
-        },
-      };
-    });
+          paginatedEntitiesOfType,
+        }),
+      },
+    };
+  });
 
+  it('view workflow in workspace from mock import', async () => {
     // Act
     await act(async () => {
       render(h(WorkflowView, { queryParams: { selectionKey } }));
     });
 
     expect(
-      screen.getByRole('button', {
+      screen.getAllByRole('button', {
         name: /inputs/i,
       })
     );
 
-    expect(screen.getByText('joint-discovery-gatk4'));
+    expect(screen.getByText('echo_to_file-configured'));
+  });
+
+  it('can run a workflow given an entity', async () => {
+    // Arrange
+    const user = userEvent.setup();
+    const namespace = 'testNamespace';
+    const name = 'testName';
+    const ws = {
+      name,
+      namespace,
+      cloudPlatform: 'Gcp',
+    };
+    const onDismiss = jest.fn();
+
+    const props = { namespace, name, ws, onDismiss };
+
+    // Act
+    await act(async () => {
+      render(h(WorkflowView, { props, queryParams: { selectionKey } }));
+    });
+
+    const selectDataButton = screen.getAllByRole('button').filter((button) => button.textContent.includes('Select Data'))[0];
+    expect(selectDataButton).toHaveTextContent('Select Data');
+
+    expect(screen.getByText('sra')).toBeInTheDocument();
+
+    const dropdown = screen.getByLabelText('Entity type selector');
+    const dropdownHelper = new SelectHelper(dropdown, user);
+    await dropdownHelper.selectOption('sra');
+    await user.click(selectDataButton);
+
+    const allSelectRadioButton = screen.getByLabelText('Select all');
+    await user.click(allSelectRadioButton);
+
+    const okButton = screen.getAllByRole('button').filter((button) => button.textContent.includes('OK'))[0];
+    await user.click(okButton);
+
+    const attributeTextbox = screen.getByRole('textbox', { name: /echo_to_file input1 attribute/i });
+    await user.type(attributeTextbox, 'this.string');
+
+    const saveButton = screen.getAllByRole('button').filter((button) => button.textContent.includes('Save'))[0];
+    await user.click(saveButton);
+
+    const runButton = screen.getAllByRole('button').filter((button) => button.textContent.includes('Run analysis'))[0];
+    await user.click(runButton);
+
+    const launchButton = screen.getAllByRole('button').filter((button) => button.textContent.includes('Launch'))[0];
+    await user.click(launchButton);
+
+    expect(Nav.goToPath).toHaveBeenCalledWith('workspace-submission-details', {
+      name: undefined,
+      namespace: undefined,
+      submissionId: undefined,
```

Comment:
Why are these undefined? It might be nice to mock out wherever these are coming from so we can assert the actual values are being passed through